jax/_src/numpy/polynomial.py

```diff
@@ -102,15 +115,131 @@ def roots(p: ArrayLike, *, strip_zeros: bool = True) -> Array:
     return _roots_with_zeros(p_arr, num_leading_zeros)
 
 
-_POLYFIT_DOC = """\
-Unlike NumPy's implementation of polyfit, :py:func:`jax.numpy.polyfit` will not warn on rank reduction, which indicates an ill conditioned matrix
-Also, it works best on rcond <= 10e-3 values.
-"""
-@implements(np.polyfit, lax_description=_POLYFIT_DOC)
 @partial(jit, static_argnames=('deg', 'rcond', 'full', 'cov'))
 def polyfit(x: Array, y: Array, deg: int, rcond: float | None = None,
             full: bool = False, w: Array | None = None, cov: bool = False
             ) -> Array | tuple[Array, ...]:
+  r"""Least squares polynomial fit to data.
+
+  Jax implementation of :func:`numpy.polyfit`.
+
+  Given a set of data points ``(x, y)`` and degree of polynomial ``deg``, the
+  function finds a polynomial equation of the form:
+
+  .. math::
+
+	   y = p(x) = p[0] x^{deg} + p[1] x^{deg - 1} + ... + p[deg]
+
+  Args:
+    x: Array of data points of shape ``(M,)``.
+    y: Array of data points of shape ``(M,)`` or ``(M, K)``.
+    deg: Degree of the polynomials.
+    rcond: Relative condition number of the fit. Default value is ``len(x) * eps``.
+    full: Switch that controls the return value. Default is ``False`` which
+      restricts the return value to the array of polynomail coefficients ``p``.
+      If ``True``, the function returns a tuple ``(p, resids, rank, s, rcond)``.
+    w: Array of weights of shape ``(M,)``. If None, all data points are considered
+      to have equal weight. If not None, the weight :math:`w_i` is applied to the
+      unsquared residual of :math:`y_i - \widehat{y}_i` at :math:`x_i`, where
+      :math:`\widehat{y}_i` is the fitted value of :math:`y_i`. Default is None.
+    cov: Boolean or string. If ``True``, returns the covariance matrix scaled
```

Comment:
We might mention that `cov` is ignored if `full` is True.

Reply:
Thanks! Updated `cov` description and squashed into single commit.